VIDEO_2_SCRIPT.md

```diff
@@ -0,0 +1,158 @@
+
+## Video 2: What kinds of API architectures exist
```

Comment:
I have a few concerns with the whole document:

1. These are not API architectures but software architectures and there are many more than those.
2. Entering the software architecture topic is complex, long, and unnecessary to learn APIs or AsyncAPI.
3. Event-driven architecture and EDA are the same things. EDA is just the acronym for Event-Driven Architecture.
4. AsyncAPI is not a kind of API architecture or software architecture but just a specification language to define APIs.

Two proposals to fix this:
1. We just mention the most common types of software architecture in which you can find APIs. Say Monolithic vs Microservices vs Serverless for instance. And we explain what role APIs play in these types of architectures.
2. We forget about architectures and go straight to types of APIs: Synchronous APIs (request/response) and Asynchronous APIs (point-to-point or broker-based).

I'd go for the second option as it's more relevant to our topic.

Reply:
My assumption was that other architectures were mentioned because it is needed to introduce event-driven architectures, because talking about AsyncAPI without EDA intro is hard.

I do agree talking about all these different architectures is complex. So I definitely propose we choose option `1` where we explain Monolith, Microservice, Serverless, what they are and then talk about EDA, that it is present in all of them. We should also explain sync vs async...might be explaining this should be first, architectures second. Most important is that to explain AsyncAPI, we need to explain EDA